release_notes/ocp-4-11-release-notes.adoc

```diff
@@ -568,6 +568,80 @@ For more information, see xref:../nodes/scheduling/nodes-descheduler.adoc#nodes-
 [id="ocp-4-11-monitoring"]
 === Monitoring
 
+The monitoring stack for this release includes the following new and modified features.
+
+==== Monitoring stack components and dependencies
+
+Updates to versions of monitoring stack components and dependencies include the following:
+
+* Alertmanager to 0.24.0
+* kube-state-metrics to 2.5.0
+* node-exporter to 1.3.1
+* prom-label-proxy to 0.4.0
+* Prometheus to 2.36.2
+* Prometheus adapter to 0.9.1
+* Prometheus operator to 0.57.0
+* Thanos to 0.26.0
+
+==== Changes to alerting rules
+
+[NOTE]
+====
+Red Hat does not guarantee backward compatibility for recording rules or alerting rules.
+====
+
+* *New*
+** Added the `KubePersistentVolumeInodesFillingUp` alert, which functions similarly to the existing `KubePersistentVolumeFillingUp` alert but is applied to inodes rather than volume space.
+** Added the `PrometheusScrapeBodySizeLimitHit` alert to detect targets hitting the body size limit.
+** Added the `PrometheusScrapeSampleLimitHit` alert to detect targets hitting the sample limit.
+
+* *Changed*
+** Fixed the `KubeDaemonSetRolloutStuck` alert to use the updated metric `kube_daemonset_status_updated_number_scheduled` from `kube-state-metrics`.
+** Replaced the `KubeJobCompletion` alert with `KubeJobNotCompleted`. The new `KubeJobNotCompleted` alert avoids false positives when an earlier job failed but the most recent job succeeded.
+** Updated the `NodeNetworkInterfaceFlapping` alert to exclude `tunbr` interfaces from the alert expression.
+** Updated the `PrometheusOperatorWatchErrors` alert to reduce the time interval from ten minutes to five minutes, to enable faster recovery of the alert, and to reduce false positives.
```

Comment:
for the recording rule `cluster:telemetry_selected_series:count`, the 4.11 expr is updated to
```
max(federate_samples - federate_filtered_samples)
```
4.10 expr is
```
count({__name__=~\"cluster:usage:.*|count:up0|count:up1|cluster_version|cluster_version_available_updates|cluster_operator_up|cluster_operator_conditions|cluster_version_payload|cluster_installer|cluster_infrastructure_provider|cluster_feature_set|instance:etcd_object_counts:sum|ALERTS|code:apiserver_request_total:rate:sum|cluster:capacity_cpu_cores:sum|cluster:capacity_memory_bytes:sum|cluster:cpu_usage_cores:sum|cluster:memory_usage_bytes:sum|openshift:cpu_usage_cores:sum|openshift:memory_usage_bytes:sum|workload:cpu_usage_cores:sum|workload:memory_usage_bytes:sum|cluster:virt_platform_nodes:sum|cluster:node_instance_type_count:sum|cnv:vmi_status_running:count|cluster:vmi_request_cpu_cores:sum|node_role_os_version_machine:cpu_capacity_cores:sum|node_role_os_version_machine:cpu_capacity_sockets:sum|subscription_sync_total|olm_resolution_duration_seconds|csv_succeeded|csv_abnormal|cluster:kube_persistentvolumeclaim_resource_requests_storage_bytes:provisioner:sum|cluster:kubelet_volume_stats_used_bytes:provisioner:sum|ceph_cluster_total_bytes|ceph_cluster_total_used_raw_bytes|ceph_health_status|job:ceph_osd_metadata:count|job:kube_pv:count|job:ceph_pools_iops:total|job:ceph_pools_iops_bytes:total|job:ceph_versions_running:count|job:noobaa_total_unhealthy_buckets:sum|job:noobaa_bucket_count:sum|job:noobaa_total_object_count:sum|noobaa_accounts_num|noobaa_total_usage|console_url|cluster:network_attachment_definition_instances:max|cluster:network_attachment_definition_enabled_instance_up:max|cluster:ingress_controller_aws_nlb_active:sum|insightsclient_request_send_total|cam_app_workload_migrations|cluster:apiserver_current_inflight_requests:sum:max_over_time:2m|cluster:alertmanager_integrations:max|cluster:telemetry_selected_series:count|openshift:prometheus_tsdb_head_series:sum|openshift:prometheus_tsdb_head_samples_appended_total:sum|monitoring:container_memory_working_set_bytes:sum|namespace_job:scrape_series_added:topk3_sum1h|namespace_job:scrape_samples_post_metric_relabeling:topk3|monitoring:haproxy_server_http_responses_total:sum|rhmi_status|cluster_legacy_scheduler_policy|cluster_master_schedulable|che_workspace_status|che_workspace_started_total|che_workspace_failure_total|che_workspace_start_time_seconds_sum|che_workspace_start_time_seconds_count|cco_credentials_mode|cluster:kube_persistentvolume_plugin_type_counts:sum|visual_web_terminal_sessions_total|acm_managed_cluster_info|cluster:vsphere_vcenter_info:sum|cluster:vsphere_esxi_version_total:sum|cluster:vsphere_node_hw_version_total:sum|openshift:build_by_strategy:sum|rhods_aggregate_availability|rhods_total_users|instance:etcd_disk_wal_fsync_duration_seconds:histogram_quantile|instance:etcd_mvcc_db_total_size_in_bytes:sum|instance:etcd_network_peer_round_trip_time_seconds:histogram_quantile|instance:etcd_mvcc_db_total_size_in_use_in_bytes:sum|instance:etcd_disk_backend_commit_duration_seconds:histogram_quantile|jaeger_operator_instances_storage_types|jaeger_operator_instances_strategies|jaeger_operator_instances_agent_strategies|appsvcs:cores_by_product:sum|nto_custom_profiles:count|openshift_csi_share_configmap|openshift_csi_share_secret|openshift_csi_share_mount_failures_total|openshift_csi_share_mount_requests_total\",alertstate=~\"firing|\",quantile=~\"0.99|0.99|0.99|\"})
```
and I don't think we need to mention it, needs @simonpasquier help to confirm

Reply:
I think `cluster:telemetry_selected_series:count` is actually a new recording rule:
```
jan@jflaptop ..rk/code/go/cluster-monitoring-operator (git)-[release-4.11] % g diff release-4.10 assets/telemeter-client/prometheus-rule.yaml
diff --git a/assets/telemeter-client/prometheus-rule.yaml b/assets/telemeter-client/prometheus-rule.yaml
new file mode 100644
index 00000000..205f42f2
--- /dev/null
+++ b/assets/telemeter-client/prometheus-rule.yaml
@@ -0,0 +1,11 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: telemetry
+  namespace: openshift-monitoring
+spec:
+  groups:
+  - name: telemeter.rules
+    rules:
+    - expr: max(federate_samples - federate_filtered_samples)
+      record: cluster:telemetry_selected_series:count

Reply:
@simonpasquier -- no change is required here, correct?

Reply:
correct